remotes/docker/config/hosts.go

```diff
@@ -530,7 +529,7 @@ func parseHostConfig(server string, baseDir string, config hostFileConfig) (host
 func getSortedHosts(root *toml.Tree) ([]string, error) {
 	iter, ok := root.Get("host").(*toml.Tree)
 	if !ok {
-		return nil, errors.New("invalid `host` tree")
+		return nil, nil
```

Comment:
At least it looks like 1.6 is impacted similarly, so would also require a patch; https://github.com/containerd/containerd/blob/d32140899a9405636ab5ae042aae81e9d3b8330e/remotes/docker/config/hosts.go#L529-L534

Given that `getSortedHost` with this patch will no longer return an error, perhaps it'd be good to change its signature to not pretend it may ever result in one;


```go
func getSortedHosts(root *toml.Tree) []string {
	iter, ok := root.Get("host").(*toml.Tree)
	if !ok {
		return nil
	}

	list := append([]string{}, iter.Keys()...)
	// go-toml stores TOML sections in the map object, so no order guaranteed.
	// We retrieve line number for each key and sort the keys by position.
	sort.Slice(list, func(i, j int) bool {
		h1 := iter.GetPath([]string{list[i]}).(*toml.Tree)
		h2 := iter.GetPath([]string{list[j]}).(*toml.Tree)
		return h1.Position().Line < h2.Position().Line
	})
	return list
}
```

Reply:
It doesn't impact main / 2.0. I tested on main branch and v2.0.0 tag. They don't have the issue.

```shell
$ sudo ctr i pull --hosts-dir /etc/containerd/certs.d docker.io/library/python:latest
Pulling from OCI Registry (docker.io/library/python:latest)     elapsed: 7.8 s  total:  233.5   (29.8 MiB/s)
$ sudo cat /etc/containerd/certs.d/docker.io/hosts.toml
server = "registry-1.docker.io"
```

Based on the comment, the behavior was gone after upgrading toml major version
https://github.com/containerd/containerd/pull/10028#issuecomment-2072438477